tests/components/hive/test_config_flow.py

```diff
@@ -266,6 +268,49 @@ async def test_option_flow(hass):
     assert result["data"][CONF_SCAN_INTERVAL] == UPDATED_SCAN_INTERVAL
 
 
+async def test_user_flow_2fa_send_new_code(hass):
+    """Resend a 2FA code if it didn't arrive."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.hive.config_flow.Auth.login",
+        return_value={
+            "ChallengeName": "SMS_MFA",
+        },
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_USERNAME: USERNAME,
+                CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["step_id"] == CONF_CODE
+    assert result2["errors"] == {}
+
+    with patch(
+        "homeassistant.components.hive.config_flow.Auth.login",
+        return_value={
+            "ChallengeName": "SMS_MFA",
+        },
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"], {CONF_CODE: MFA_RESEND_CODE}
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result3["step_id"] == CONF_CODE
+    assert result3["errors"] == {}
```

Comment:
Please complete the flow in each test to either an error, an abort or a create entry.